applications/ShallowWaterApplication/custom_utilities/post_process_utilities.h

```diff
@@ -0,0 +1,311 @@
+        {
+            auto prop_id = (mrModelPart.PropertiesBegin() + i)->Id();
+            max_id = std::max(max_id, prop_id);
+            prop_ids.push_back(prop_id);
+        }
+
+        // Creating the auxiliary ids for the dry domain
+        for (auto id : prop_ids)
+        {
+            // Get pointers to the properties and create the dry property
+            Properties::Pointer wet_prop = mrModelPart.pGetProperties(id);
+            Properties::Pointer dry_prop(new Properties(*wet_prop));
+            dry_prop->SetId(++max_id);
+
+            // Add the new properties to the model part and add them to the maps
+            mrModelPart.AddProperties(dry_prop);
+            mWetToDryPropertiesMap[wet_prop->Id()] = max_id;
+            mDryToWetPropertiesMap[dry_prop->Id()] = id;
+        }
+    }
+
+    void AssignDryWetProperties(Flags Flag)
+    {
+        #pragma omp parallel for
+        for (int i = 0; i < static_cast<int>(mrModelPart.NumberOfElements()); ++i)
+        {
+            auto elem = mrModelPart.ElementsBegin() + i;
+
+            if (elem->Is(Flag))
+            {
+                auto search = mDryToWetPropertiesMap.find(elem->GetProperties().Id());
+                if (search != mDryToWetPropertiesMap.end()) // The element was dry
+                {
+                    IndexType wet_prop_id = search->second;
+                    elem->SetProperties(mrModelPart.pGetProperties(wet_prop_id));
+                }
+            }
+            else
+            {
+                auto search = mWetToDryPropertiesMap.find(elem->GetProperties().Id());
+                if (search != mWetToDryPropertiesMap.end()) // The element was wet
+                {
+                    IndexType dry_prop_id = search->second;
+                    elem->SetProperties(mrModelPart.pGetProperties(dry_prop_id));
+                }
+            }
+        }
+    }
+
+    void RestoreDryWetProperties()
+    {
+        #pragma omp parallel for
+        for (int i = 0; i < static_cast<int>(mrModelPart.NumberOfElements()); ++i)
+        {
+            auto elem = mrModelPart.ElementsBegin() + i;
+
+            auto search = mDryToWetPropertiesMap.find(elem->GetProperties().Id());
+            if (search != mDryToWetPropertiesMap.end()) // The element was dry
+            {
+                IndexType wet_prop_id = search->second;
+                elem->SetProperties(mrModelPart.pGetProperties(wet_prop_id));
+            }
+        }
+    }
+
+    ///@}
+    ///@name Access
+    ///@{
+
+
+    ///@}
+    ///@name Inquiry
+    ///@{
+
+
+    ///@}
+    ///@name Input and output
+    ///@{
+
+    /// Turn back information as a string.
+    std::string Info() const;
+
+    /// Print information about this object.
+    void PrintInfo(std::ostream& rOStream) const;
+
+    /// Print object's data.
+    void PrintData(std::ostream& rOStream) const;
+
+
+    ///@}
+    ///@name Friends
+    ///@{
+
+
+    ///@}
+
+protected:
```

Comment:
You can remove the protected section if you don't need it to make the file shorter.

Reply:
Done